packages/inbound-email-handler/src/index.ts

```diff
@@ -42,51 +42,60 @@ export const getNewsletterHandler = (
 
 export const inboundEmailHandler = Sentry.GCPFunction.wrapHttpFunction(
   async (req, res) => {
-    const parts = multipart.parse(req.body, 'xYzZY')
-    const parsed: Record<string, string> = {}
-
-    let pdfAttachment: Buffer | undefined
-    let pdfAttachmentName: string | undefined
-
-    for (const part of parts) {
-      const { name, data, type, filename } = part
-      if (name && data) {
-        parsed[name] = data.toString()
-      } else if (type === 'application/pdf' && data) {
-        pdfAttachment = data
-        pdfAttachmentName = filename
-      } else {
-        console.log('no data or name for ', part)
-      }
-    }
-
-    const headers = parseHeaders(parsed.headers)
-    console.log('parsed: ', parsed)
-    console.log('headers: ', headers)
-
     try {
-      const from = parsed.from
-      const subject = parsed.subject
-      const html = parsed.html
-      const text = parsed.text
+      const parts = multipart.parse(req.body, 'xYzZY')
+      const parsed: Record<string, string> = {}
+
+      let pdfAttachment: Buffer | undefined
+      let pdfAttachmentName: string | undefined
+
+      for (const part of parts) {
+        const { name, data, type, filename } = part
+        if (name && data) {
+          parsed[name] = data.toString()
+        } else if (type === 'application/pdf' && data) {
+          pdfAttachment = data
+          pdfAttachmentName = filename
+        } else {
+          console.log('no data or name for ', part)
+        }
+      }
 
-      const forwardedAddress = headers['x-forwarded-to']?.toString()
-      const recipientAddress = forwardedAddress || parsed.to
+      const headers = parseHeaders(parsed.headers)
+      console.log('parsed: ', parsed)
+      console.log('headers: ', headers)
+
+      // original sender email address
+      const from = headers['from'].toString()
+      const subject = headers['subject']?.toString()
+      const html = parsed['html']
+      const text = parsed['text']
+
+      // headers added when forwarding email by some rules in Gmail
+      // e.g. 'X-Forwarded-To: recipient@omnivore.app'
+      const forwardedTo = headers['x-forwarded-to']?.toString().split(',')[0]
+      // x-forwarded-for is a space separated list of email address
+      // the first one is the forwarding email sender and the last one is the recipient
+      // e.g. 'X-Forwarded-For: sender@omnivore.app recipient@omnivore.app
+      const forwardedFrom = headers['x-forwarded-for']?.toString().split(' ')[0]
```

Comment:
I guess this is the key to the whole thing here.

Reply:
Yeah, I have updated the description.
To sum up, the issue is Gmail changes the `from` to the original sender email address instead of the Omnivore account email address.